### ISSUE TITLE
Undefined array key "candidatesTokenCount" when streaming through Gemini provider

### ISSUE THREAD

**Issue by tg666** [inspector-apm/neuron-ai#224]
**Describe the bug**

When using the `stream` method from the AI provider `Gemini`, the following warning is reported:

```
Warning: Undefined array key "candidatesTokenCount" in /var/www/html/vendor/inspector-apm/neuron-ai/src/Providers/Gemini/HandleStream.php on line 55
```

While the `promptTokenCount` key probably exists in all events of a given stream, the `candidatesTokenCount` key is found only in the last one.

**To Reproduce**

```php
<?php

declare(strict_types=1);

use NeuronAI\Providers\Gemini\Gemini;
use NeuronAI\Agent;
use NeuronAI\Chat\Messages\UserMessage;

require __DIR__ . '/vendor/autoload.php';

$gemini = new Gemini(
    key: '<API_KEY>',
    model: 'gemini-2.0-flash',
);

$agent = new Agent()->setAIProvider($gemini);

foreach ($agent->stream(new UserMessage('Hello world!')) as $output) {
    echo $output;
}

echo "\n";
```

**Expected behavior**

There should be no warning.

I think that `usage` should only be yielded on the last event (once the `finishReason` key exists), since the implementation in `NeuronAI\HandleStream` sums all yielded usages.

**The Neuron AI version you are using:**
 - Version 1.15.8 (latest)

**Additional context**

Emitted stream events from the above example:

```sh
{"candidates":[{"content":{"parts":[{"text":"Hello"}],"role":"model"}}],"usageMetadata":{"promptTokenCount":19,"totalTokenCount":19,"promptTokensDetails":[{"modality":"TEXT","tokenCount":19}]},"modelVersion":"gemini-2.0-flash","responseId":"AABvaJHYDIeXhMIPzbCKuAw"}
{"candidates":[{"content":{"parts":[{"text":" there! How"}],"role":"model"}}],"usageMetadata":{"promptTokenCount":19,"totalTokenCount":19,"promptTokensDetails":[{"modality":"TEXT","tokenCount":19}]},"modelVersion":"gemini-2.0-flash","responseId":"AABvaJHYDIeXhMIPzbCKuAw"}
{"candidates":[{"content":{"parts":[{"text":" can I help you today?\n"}],"role":"model"},"finishReason":"STOP"}],"usageMetadata":{"promptTokenCount":18,"candidatesTokenCount":11,"totalTokenCount":29,"promptTokensDetails":[{"modality":"TEXT","tokenCount":18}],"candidatesTokensDetails":[{"modality":"TEXT","tokenCount":11}]},"modelVersion":"gemini-2.0-flash","responseId":"AABvaJHYDIeXhMIPzbCKuAw"
```

**Comment by ilvalerione** [comment]
Fixed in [1.15.9](https://github.com/inspector-apm/neuron-ai/releases/tag/1.15.9)

Thank you for your report.
